### PR TITLE
Added `Graph` method `max_rigid_dimension`

### PR DESCRIPTION
**Pull request by matthiashimmelmann**
The handles Tony's request about the maximum rigid dimension.

### CODE REVIEW

File: pyrigi/graph.py [hunk 1]
```diff
@@ -1937,6 +1937,45 @@ def min_rigid_subgraphs(self, dim: int = 2) -> List[Graph]:
...
+        3
+        """
+        if nx.number_of_selfloops(self) > 0:
+            raise LoopError()
+
+        V = self.number_of_nodes()
+        E = self.number_of_edges()
+        if E == V * (V - 1) / 2:
+            return math.inf
+        # Find the larger root of the Maxwell function E = d*V-d*(d+1)/2
```
Comment (MatteoGallet) [inline pyrigi/graph.py:1971]: What about 

```
# Find the largest d such that d*(d+1)/2 - d*V + E = 0
```

?

Comment (matthiashimmelmann) [reply pyrigi/graph.py:1971]: I was wondering if we can do `floor` as well here? I guess it doesn't save much, just a small nitpick.

Comment (Legersky) [reply pyrigi/graph.py]: Yes, I think we can.
We can also directly output 0 if the graph is disconnected.

**Comment by Legersky** [comment]
I am thinking whether we should use `sympy.oo` instead of `math.inf` to have a "symbolic" infinity rather than float?

File: pyrigi/graph.py [hunk 2]
```diff
@@ -1937,6 +1937,51 @@ def min_rigid_subgraphs(self, dim: int = 2) -> List[Graph]:
                         rigid_subgraphs[H2] = False
         return [list(H) for H, is_min in rigid_subgraphs.items() if is_min]
 
+    @doc_category("Generic rigidity")
+    def max_rigid_dimension(self) -> int | Inf:
+        """
+        Compute the maximum dimension, in which a graph is generically rigid.
```
Comment (MatteoGallet) [inline pyrigi/graph.py:1943]: Could you please link to the math doc?

**Review by MatteoGallet — APPROVED** [review]

**Pull request merged**